### PR TITLE
docs(theming): add section about safe area padding

### PR DESCRIPTION
**Pull request by amandaejohnston**
Shortcut: https://ionic-docs-git-fw-4712-ionic1.vercel.app/docs/theming/advanced#safe-area-padding

### CODE REVIEW

File: docs/theming/advanced.md [hunk 1]
```diff
@@ -180,3 +180,29 @@ However, because CSS variables can be set at runtime and are more dynamic, it is
...
+
+For example, below are screenshots of an iPhone 14 Pro Max. The red section is the safe area, and the white sections are places where the app's content would be covered up.
+
+| Portrait                                                                       | Landscape                                                                        |
+| ------------------------------------------------------------------------------ | -------------------------------------------------------------------------------- |
+| <img src={require('@site/static/img/theming/ios-portrait-top.png').default} /> | <img src={require('@site/static/img/theming/ios-landscape-left.png').default} /> |
+
+To accommodate this, Ionic automatically adds padding to certain components. For example, an `ion-header` component placed in an `ion-modal` will receive padding according to the top edge of the device's safe area. This avoids the device's notch covering up the header text.
```
Comment (liamdebeasi) [inline docs/theming/advanced.md:194]: Nit: We apply the safe area to the first toolbar inside the header, not the header itself

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

**Comment by brandyscarney** [comment]
Great description! Is it possible to crop out the background behind the device? It is a little jarring in dark mode:

Screenshot 2023-07-31 at 2 45 15 PM

**Review by liamdebeasi — APPROVED** [review]

**Comment by amandaejohnston** [comment]
@brandyscarney Good call, I trimmed it out 👍 I left the modal padding screenshot alone since I think the white background looks okay there, but LMK if you'd like that removed too.

**Review by brandyscarney — APPROVED** [review]
I think the modal one is fine! This looks really good now. 👍

**Pull request merged**